### PR TITLE
Connection Thread to Test Server 

### LINKED ISSUE #95: Establish Connection to Relay from Test Server
Spawn a thread that runs a connection from relay to a test server to stand in the place of iMotions.

### PR DESCRIPTION
**Pull request by carlyrhoover**
### Fixes #95

---

### **What was changed?**
-Thread created in main.rs of relay 
-Connection message enum added in channel.rs
---

### **Why was it changed?**
-establishes connection to test server 

---

### **How was it changed?**
-Changes in src of relay

---

### **Screenshots (if applicable):**
connectedmeow

### CODE REVIEW

File: relay/src/main.rs [hunk 1]
```diff
@@ -20,6 +26,31 @@ use interprocess::local_socket::{
 };
 
 fn main() -> iced::Result {
+     //tcp connection 
+    
+    // Connect to the server 
+
+    let (send, recv) = std::sync::mpsc::channel::<channelMessage>();
+    let tcp_onnection = thread::spawn(move || {
```
Comment (hrmeiners) [inline relay/src/main.rs:34]: 😭 I think this is supposed to be `let tcp_connection = ...`, right? I think you forgot the 'c'

File: relay/src/main.rs [hunk 2]
```diff
@@ -20,6 +26,31 @@ use interprocess::local_socket::{
...
+        match TcpStream::connect("127.0.0.1:7878"){   
+            Ok(mut stream) => {
+                println!("Successfull connected.");
+                let message = channelMessage::CONNECT;
+                send.send(message);
+        }
+
+            Err(e) => {
+                println!("Connection failed: {}", e);
```
Comment (hrmeiners) [inline relay/src/main.rs:43]: Were you going to implement a TCP retry mechanism this sprint too? Or am I misremembering and getting my wires crossed?

**Review by hrmeiners — DISMISSED** [review]
I am a little bit confused on where this code fits into what we demo'd with the client today, as it only seems to send one connection message/response when we demo'd a continuous stream. I think I'm missing something here, but I don't see anything explicitly wrong or incorrect with the code you wrote and am glad it successfully connected!!

File: relay/src/main.rs [hunk 3]
```diff
@@ -20,6 +26,31 @@ use interprocess::local_socket::{
...
+        }
+
+
+}
+
+     
+
```
Comment (aelnourSLU) [inline relay/src/main.rs:50]: You can use `cargo fmt` to format the document before pushing, it helps to keep layout standard.

File: relay/src/main.rs [hunk 4]
```diff
@@ -20,6 +26,31 @@ use interprocess::local_socket::{
...
 fn main() -> iced::Result {
+     //tcp connection 
+    
+    // Connect to the server 
+
+    let (send, recv) = std::sync::mpsc::channel::<channelMessage>();
+    let tcp_onnection = thread::spawn(move || {
+        match TcpStream::connect("127.0.0.1:7878"){   
+            Ok(mut stream) => {
+                println!("Successfull connected.");
```
Comment (aelnourSLU) [inline relay/src/main.rs:37]: Small typo in print output here

File: relay/src/channel.rs [hunk 5]
```diff
@@ -0,0 +1,3 @@
+pub enum channelMessage{
```
Comment (aelnourSLU) [inline relay/src/channel.rs:1]: Rust convention is to use Upper Camel Case, so `ChannelMessage` is preferred here

File: relay/src/channel.rs [hunk 6]
```diff
@@ -0,0 +1,3 @@
+pub enum channelMessage{
+    CONNECT,
```
Comment (aelnourSLU) [inline relay/src/channel.rs:2]: Likewise, enum variants conventionally start with a capital letter and are lowercase for the rest, so `Connect` is preferred here

**Review by aelnourSLU — COMMENTED** [review]
This looks like a very solid start. I have small formatting suggestions, but none are critical for code functionality.

**Review by DeviPriya-Kolla — APPROVED** [review]
Thank you for addressing issue #95 in this PR. For future iterations, consider expanding test coverage to capture additional edge cases, which can help prevent regressions and ensure robustness.

**Review by hrmeiners — APPROVED** [review]
